.github/actions/github-list-changes/action.yml

```diff
@@ -0,0 +1,26 @@
+name: "Github List Changes"
+description: "List all changed files in a pushed set of commits or a pull request"
+inputs:
+  write-list-to-env:
+    description: "If set to true then write the list of changed files to the environment variable GITHUB_MODIFIED_FILES"
+    default: "false"
+outputs:
+  all_changed_files:
+    description: The list of changed files (one per line)
+    value: ${{ steps.list-changes.outputs.all-changed-files }}
+runs:
+  using: "composite"
+  steps:
+    - name: "Checkout"
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: 0
```

Comment:
embedding checkout is usually not a wise choice because it would override any previous customization

if you need the `detch-depth: 0` for things to work, then add it  in the readme in the example snippet

Reply:
Ah - ok. The `fetch-depth: 0` was exactly the reason to include this.